src/ngx_http_lua_args.c

```diff
@@ -544,7 +542,109 @@ ngx_http_lua_ffi_req_get_uri_args(ngx_http_request_t *r, u_char *buf,
 
     return i;
 }
-#endif /* NGX_LUA_NO_FFI_API */
 
 
+int
+ngx_http_lua_ffi_req_get_uri_args(ngx_http_request_t *r, u_char *buf,
+    ngx_http_lua_ffi_table_elt_t *out, int count)
+{
+    ngx_memcpy(buf, r->args.data, r->args.len);
+
+    return ngx_http_lua_ffi_req_args_helper(buf, r->args.len, out, count);
+}
+
+
+
+int
+ngx_http_lua_ffi_req_get_post_args_len(ngx_http_request_t *r)
+{
+    size_t       len;
+    ngx_chain_t *cl;
+
+    if (r->connection->fd == (ngx_socket_t) -1) {
+        return NGX_HTTP_LUA_FFI_BAD_CONTEXT;
+    }
+
+    if (r->discard_body || r->request_body->bufs == NULL) {
+        return -1;
+    }
+
+    if (r->request_body == NULL) {
+        return -2;
+    }
+
+    if (r->request_body->temp_file) {
+        return -3;
```

Comment:
We should avoid magical numbers like -1, -2, and -3.

Reply:
@agentzh you asked me to avoid constants in this comment: https://github.com/openresty/lua-nginx-module/pull/985#discussion_r117308957

So what is the appropriate design in your eyes?